api/proto/info/info.proto

```diff
@@ -504,9 +506,44 @@ message ZInfoNetworkInstance {
     ZInfoVpn vinfo = 31;
     ZInfoLisp linfo = 32;
   }
+  UsageInfo usage = 33;
   repeated ErrorInfo networkErr = 40;
 }
 
+message UsageInfo {
+  google.protobuf.Timestamp createTime = 1;
+  uint32 refCount = 2;
+  google.protobuf.Timestamp lastUseTime = 3; // When refCount dropped to zero
+}
+
+message VolumeResources {
+  uint64 maxSizeBytes = 1; // From config
+  uint64 curSizeBytes = 2; // Current disk usage
+}
+
+message ZInfoVolume {
+  string uuid = 1;
+  string displayName = 2;
+  UsageInfo usage = 3;
+  VolumeResources resources = 4;
+  ZSwState state = 5;
+  uint32 downloadProgress = 6;
+
+  repeated ErrorInfo volumeErr = 7;
```

Comment:
nit.. should we just call it errors ( as opposed to volumeErr ) - seems redundant..

Also - would it be good to have an ErrorInfowrapper for this array that other objects can also use? That would allow code to be shared for processing this ErrorInfo..

message ErrorInfoArray {
   repeated ErrorInfo errInfo = 1
}

If we standardize this as the way to pass errors for various objects, it would simplify code..
The existing objects can be moved to use this slowly..

Reply:
@kalyan-nidumolu I don't see how the device would produce a sequence of errors for a volume. The processing stops on the first error.

Reply:
@bharani-zededa did you have a use case where you would need to have a sequence of errors?

Reply:
@eriknordmark No, I just borrowed it from networkInstance. Guess networkInstance is handled differently form volume and so it should be singleton in volume. I will change it.